### DISCUSSION TITLE
Get Full DB Layout From PLC

### DISCUSSION THREAD

**Discussion by DavoedM** [gijzelaerr/python-snap7#400] [Q&A] [answered]
Hello,

I can see how to create a layout for a known DB in the util example section. What I would like to be able to do is to create a layout, even if it does not have variable names from info gathered from the PLC.

Is this possible? TIA can obviously get this data, including vaiable names comments etc from a PLC, but is it possible to do this with Snap7/python-snap7?

Thanks,
David

**Comment by spreeker** [comment, marked as answer]
I'm pretty sure TIA does not get variable names / comments directly from the PLC all this information is NOT stored on the PLC. Variable names comments are stored in de project files of siemens /  TIA reads those.  PLC's have very little memory. Snap7 cannot read this information because it is and never will be stored on the PLC.

**Reply by xx1raven1xx** [reply]
This statement is true for controllers of the S7 300 and S7 400 family. S7 1500 controllers can store the names of variables in themselves.

**Reply by mateusz-glowinski** [reply]
@xx1raven1xx Can i somehow get the names using this package for S7-1500 family?

**Comment by gijzelaerr** [comment]
Closing as answered. Variable names and comments are not stored on the PLC -- they exist only in the TIA Portal project files.
